### PR TITLE
Native forgejo support

### LINKED ISSUE #3637: ForgeJo icon for "Go to forge page" button
### Clear and concise description of the problem

If you visit a repository page on Woodpecker for a project hosted on a ForgeJo instance, the "Go to forge" button between the pipeline badge and the "Settings" button is Gitea's icon.

Reference:
https://codeberg.org/Codeberg-CI/feedback/issues/177

### Suggested solution

I think it should be Forgejo's, as it now a GitTea hard fork.

### Alternative

A default, anonymous icon.

### Additional context

_No response_

### Validations

- [X] Checked that the feature isn't part of the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by qwerty287**
I found a third-party fork of the gitea SDK: https://codeberg.org/mvdkleijn/forgejo-sdk

Closes #3637

### CODE REVIEW

File: cmd/server/flags.go [hunk 1]
```diff
@@ -372,6 +372,19 @@ var flags = append([]cli.Flag{
...
 	//
+	// Forgejo
+	//
+	&cli.BoolFlag{
+		EnvVars: []string{"WOODPECKER_FORGEJO"},
+		Name:    "forgejo",
+		Usage:   "forgejo driver is enabled",
+	},
+	&cli.StringFlag{
+		EnvVars: []string{"WOODPECKER_DEV_FORGEJO_OAUTH_URL"},
```
Comment (xoxys) [inline cmd/server/flags.go:383]: Planned to be deprecated for gitea already should not be added.

File: docs/docs/30-administration/11-forges/35-forgejo.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
+---
+toc_max_heading_level: 2
+---
+
+# Forgejo
+
+Woodpecker comes with built-in support Forgejo. To enable Forgejo you should configure the Woodpecker container using the following environment variables:
```
Comment (xoxys) [inline docs/docs/30-administration/11-forges/35-forgejo.md:7]:
```suggestion
Woodpecker comes with built-in support for Forgejo. To enable Forgejo you should configure the Woodpecker container using the following environment variables:
```

**Comment by xoxys** [comment]
Nice work. Im a bit unsure about the sdk. Is it stable enough to rely on it? Does it anything different from the gitea sdk? As long as forgejo is api compatible we could still use the gitea api.

**Comment by 6543** [comment]
forgejo as to be added to cSpell list

**Review by jolheiser — COMMENTED** [review]
just a few nits

**Comment by pat-s** [comment]
So man changes 👀 

Maybe worth letting the maintainer know that his fork is now being used by WP?

**Comment by mvdkleijn** [comment]
Hiya, maintainer of the forgejo sdk here..

Just wanted to let you know that the SDK is indeed a fork of the Gitea SDK. I made the fork after reading about Forgejo's hardfork since I thought that would make sense.

Right now, the fork is the same as the original. I even pulled in the latest updates not too long ago.

I plan to add convenience functions next to functions referencing the Forgejo api 1-on-1.

In general during the course of 1.x you can expect bug fixes, additions and cleanups. The SDK will use semver so **if** I make backwards incompatible changes, you can expect a major version bump.

I'll probably also add a vanity url for the SDK that people can optionally use so I can move the repo to a different forge if I ever need / want to.

My first order of business though is to make a woodpecker pipeline. I already setup WP locally to test the pipeline since I am new to WP.

Feel free to leave any feedback over at the repo.  (sorry for temporarily hijacking this PR with my comment :-p )

File: .cspell.json [hunk 3]
```diff
@@ -148,7 +148,8 @@
     "xormigrate",
     "xyaml",
     "zerolog",
-    "zerologger"
+    "zerologger",
+    "Forgejo"
```
Comment (pat-s) [inline .cspell.json:152]: Nitpick: The previous items look sorted - maybe align the new entry?

File: docs/docs/30-administration/11-forges/35-forgejo.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+## Forgejo on the same host with containers
+
+If you have Forgejo also running on the same host within a container, make sure the agent does have access to it.
+The agent tries to clone using the URL which Forgejo reports through its API. For simplified connectivity, you should add the Woodpecker agent to the same docker network as Forgejo is in.
+Otherwise, the communication should go via the `docker0` gateway (usually 172.17.0.1).
+
+To configure the Docker network if the network's name is `forgejo`, configure it like this:
+
+```diff title="docker-compose.yaml"
+ version: '3'
```
Comment (pat-s) [inline docs/docs/30-administration/11-forges/35-forgejo.md:25]: docker-compose yaml versions are deprecated and cosmetic. I'd suggest to leave them out.

**Review by pat-s — APPROVED** [review]
Thanks for doing all the work, @qwerty287!

Even though you tested it, maybe we should add a note in the release notes that this is "experimental" for now?

**Pull request merged**